pkg/workload/workload.go

```diff
@@ -261,15 +262,26 @@ func dropExcludedResources(input corev1.ResourceList, excludedPrefixes []string)
 	return res
 }
 
+// IsUsingTAS returns information if the workload is using TAS
+func (i *Info) IsUsingTAS() bool {
+	return slices.ContainsFunc(i.TotalRequests,
+		func(ps PodSetResources) bool {
+			return ps.TopologyRequest != nil
+		})
+}
+
 // TASUsage returns topology usage requested by the Workload
-func (i *Info) TASUsage() []TopologyDomainRequests {
-	if !features.Enabled(features.TopologyAwareScheduling) {
-		return nil
-	}
-	result := make([]TopologyDomainRequests, 0)
+func (i *Info) TASUsage() map[kueue.ResourceFlavorReference][]TopologyDomainRequests {
+	result := make(map[kueue.ResourceFlavorReference][]TopologyDomainRequests, 0)
 	for _, ps := range i.TotalRequests {
 		if ps.TopologyRequest != nil {
-			result = append(result, ps.TopologyRequest.DomainRequests...)
+			psFlavors := sets.New[kueue.ResourceFlavorReference]()
+			for _, psFlavor := range ps.Flavors {
+				psFlavors.Insert(psFlavor)
+			}
```

Comment:
```suggestion
			psFlavors := sets.New[kueue.ResourceFlavorReference](ps.Flavors)
```

Reply:
Ah, utilmaps.Values does not work as is does not deduplicate. We need "DistinctValues" (good that the new test picked it up), but not sure it warrants a new helper function. The suggestion from your comment does not compile. 
For now returning to how it was.

Reply:
reverted